src/AzureFunctions/Example/GraphQLExecutionResult.cs

```diff
@@ -0,0 +1,46 @@
+// <copyright file="GraphQLExecuterExtensions.cs">
+// MIT License, taken from https://github.com/tpeczek/Demo.Azure.Functions.GraphQL/blob/master/Demo.Azure.Functions.GraphQL/Infrastructure/GraphQLExecutionResult.cs
+// </copyright>
+// <author>https://github.com/tpeczek</author>
+using GraphQL;
+using GraphQL.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Example
+{
+    internal class GraphQLExecutionResult : ActionResult
+    {
+        private const string CONTENT_TYPE = "application/json";
+
+        private readonly ExecutionResult _executionResult;
+
+        public GraphQLExecutionResult(ExecutionResult executionResult)
+        {
+            _executionResult = executionResult ?? throw new ArgumentNullException(nameof(executionResult));
+        }
+
+        public override async Task ExecuteResultAsync(ActionContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            IDocumentWriter documentWriter = context.HttpContext.RequestServices.GetRequiredService<IDocumentWriter>();
+
+            HttpResponse response = context.HttpContext.Response;
+            response.ContentType = CONTENT_TYPE;
+            response.StatusCode = StatusCodes.Status200OK;
+
+            using var stream = new MemoryStream();
+            await documentWriter.WriteAsync(stream, _executionResult);
```

Comment:
Why not to write _executionResult directly into `response.Body` ?

Reply:
Then we run into the dotnet 3 async IO Problem. When this is solved with the migration to System.Text.Json, this can be done directly.

Reply:
For the umpteenth time I forget it. Do you leave a comment about this?

Reply:
Makes totally sense. It’s now there.